CONTRIBUTING.md

```diff
@@ -29,29 +32,75 @@ paragraphs, each separated by a blank line. It should explain the problem you so
 
 For example:
 
+---
 ```
 :sparkles: feat(etl): add new data source for customer segmentation
 ```
-
+---
 If you have multiple commits in one merge request, you should squash them into a single commit with a suitable commit
 message when merging with the main branch. The commit message should summarize what changes you made in the merge
 request and why. It should also reference the issue that you resolved and include the ticket number from Jira. For
 example:
 
+---
 ```
 :recycle: refactor(ansible): move dev machines group vars to the role vars #123
 ```
-
+---
 This indicates that you resolved ticket number 123, which was about refactoring the variables of development machines.
 
 
 ## Branching Name Convention :twisted_rightwards_arrows:
 
 Before creating a feature or resolving a bug, it is required to create an issue on GitHub. This issue will be assigned a number, which should be included in the branch name, separated by hyphens (-). Here is an example of a valid Git branch name:
+---
+---
 ```bash
 2-feature-add-contribution-documentation
 ```
+---
+## Git Rebase Policy :rotating_light:
+
+We follow a Git rebase policy to keep our Git history clean and easy to understand. When you are working on a feature or bug fix, you should create a feature branch based on the latest main branch. Then, you should make your changes in this branch and commit them as usual. Once you are done with your changes and ready to merge your branch into the main branch, you should rebase your branch on the latest main branch to make sure it is up to date and has no conflicts.
+
+#### Git Merge vs Git Rebase :zap:
+
+- We use Git rebase and Git merge to integrate changes from feature branches into the main branch. Git merge combines the changes from the feature branch into the main branch as a new commit, while Git rebase applies the changes from the feature branch onto the main branch as if they were made directly on the main branch. Both approaches have their benefits and drawbacks.
+
+- Git merge is a simple and straightforward approach that preserves the entire history of the feature branch, making it easier to understand and revert changes. However, it can create a messy and cluttered Git history if not used properly, as each merge commit adds another layer of history.
+
+- Git rebase, on the other hand, creates a cleaner and more linear Git history by applying the changes from the feature branch onto the main branch directly. This approach makes it easier to understand the relationship between different commits and can help avoid merge conflicts. However, it can be more difficult to revert changes and can cause conflicts if multiple people are working on the same feature branch.
+
+![Alt text](git-merge-vs-rebase.jpg)
```

Comment:
Please create a directory named `assets` or any related name (like `static_files`) to the images and other artifacts that may be added to the repository.

Please replace the `Alt text` with something meaningful. It means alternative text, which is shown whenever the image can not be loaded for any reason.